cmd/installer/cli/join_printcommand.go

```diff
@@ -3,18 +3,39 @@ package cli
 import (
 	"context"
 	"fmt"
+	"os"
 
 	"github.com/replicatedhq/embedded-cluster/cmd/installer/kotscli"
 	"github.com/replicatedhq/embedded-cluster/pkg/runtimeconfig"
+	rcutil "github.com/replicatedhq/embedded-cluster/pkg/runtimeconfig/util"
 	"github.com/spf13/cobra"
 )
 
 func JoinPrintCommandCmd(ctx context.Context, name string) *cobra.Command {
+	var rc runtimeconfig.RuntimeConfig
+
 	cmd := &cobra.Command{
 		Use:   "print-command",
 		Short: fmt.Sprintf("Print controller join command for %s", name),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if os.Getuid() != 0 {
+				return fmt.Errorf("update command must be run as root")
+			}
+
+			var err error
+			rc, err = rcutil.GetRuntimeConfigFromCluster(ctx)
+			if err != nil {
+				return fmt.Errorf("failed to init runtime config from cluster: %w", err)
+			}
+
+			os.Setenv("TMPDIR", rc.EmbeddedClusterTmpSubDir())
+
+			return nil
```

Comment:
Curious, is this only required now? Or was it already being done before but in a different place?